platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r812/CustomTestEventsCrossVersionSpec.groovy

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r812
+
+import groovy.transform.CompileStatic
+import org.gradle.integtests.tooling.TestLauncherSpec
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.TestLauncher
+import org.gradle.tooling.events.OperationDescriptor
+import org.gradle.tooling.events.task.TaskOperationDescriptor
+import org.gradle.tooling.events.test.TestOperationDescriptor
+
+@ToolingApiVersion('>=8.8')
```

Comment:
Why would this be limited to TAPI clients 8.8+?

Reply:
Inherited from the class I copied from. I'll do a deeper check on this.